### ISSUE TITLE
SIGSEGV with Cooperative GC 

### ISSUE THREAD

**Issue by jaykrell** [mono/mono#7098]
Cooperative GC failing in master?

Notice --with-cooperative-gc=yes.
Which I noticed in Jenkins.

Just me?
```
$ ./configure -disable-boehm -disable-nls -disable-system-aot -prefix /inst/m2 CFLAGS=-g -O0 --with-cooperative-gc=yes

MONO_PATH=mcs/class/lib/net_4_x-darwin/ mono/mini/mono mono/tests/bug-58782-plain-throw.exe

   frame #0: 0x00007fff9f3093ee libsystem_kernel.dylib`__wait4 + 10
   frame #1: 0x00000001038a1dcf mono`mono_handle_native_crash(signal="SIGSEGV", ctx=0x00000001048e8f48, info=0x00000001048e8ee0) at mini-exceptions.c:2737
   frame #2: 0x00000001037857b7 mono`mono_sigsegv_signal_handler(_dummy=11, _info=0x00000001048e8ee0, context=0x00000001048e8f48) at mini-runtime.c:3091
   frame #3: 0x00007fff9f3e9b3a libsystem_platform.dylib`_sigtramp + 26
   frame #4: 0x000000010389fafb mono`mono_handle_exception_internal(ctx=0x0000700008890998, obj=0x0000000104402018, resume=0, out_ji=0x0000000000000000) at mini-exceptions.c:1887
   frame #5: 0x000000010389f91a mono`mono_handle_exception(ctx=0x0000700008890998, obj=0x0000000104402018) at mini-exceptions.c:2363
   frame #6: 0x0000000103990557 mono`mono_amd64_throw_exception(dummy1=4366278680, dummy2=27, dummy3=4366278656, dummy4=0, dummy5=0, dummy6=0, mctx=0x0000700008890c30, exc=0x0000000104402018, rethrow=1) at exceptions-amd64.c:401
   frame #7: 0x0000000103f4f927
   frame #8: 0x0000000103aae7d8 mono`ftnptr_eh_callback_default(gchandle=27) at marshal.c:12442
   frame #9: 0x0000000103a9050a mono`mono_marshal_ftnptr_eh_callback(gchandle=27) at marshal.c:12427
   frame #10: 0x00000001043e36a3
   frame #11: 0x0000000104899317 libtest.0.dylib`mono_test_native_to_managed_exception_rethrow_thread(arg=0x00000001043e3510) at 
=================================================================
Got a SIGSEGV while executing native code. This usually indicates```
```

Part of #7046

**Comment by marek-safar** [comment]
@jaykrell could you add Jenkins log

**Comment by jaykrell** [comment]
I noticed the configure line in jenkins. The failure is local.

**Comment by lambdageek** [comment]
This is one of the known crashes on the [master coop lane](https://jenkins.mono-project.com/job/test-mono-mainline-coop/) that should be fixed as part fo #6921

**Comment by luhenry** [comment]
Duplicate of https://github.com/mono/mono/issues/6921
